src/app/components/CustomNetworkList/CustomNetworkList.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+
+import style from './CustomNetworkList.css'
+
+class CustomNetworkList extends Component {
+  delete = (index) => {
+    const { deleteCustomNetwork, setNetwork, network } = this.props
+
+    // If this is the current network they're using, reset to MainNet (index 0)
+    if (network.id === index) {
+      setNetwork(0)
+    }
+
+    deleteCustomNetwork(index)
+  }
+
+  render() {
+    const { networks } = this.props
+
+    const networkRows = []
+
+    networks.forEach((network, index) => {
```

Comment:
It might be better to keep this kind of logic outside of the render function.

Reply:
You're saying just throw it in a different method of the same class?

Reply:
Yes, render functions are ideally just template expressions. All of the generation of children are best abstracted out to their own methods for easy testing and clear understanding of their functionality.